### ISSUE TITLE
Invalid composite key with Android 12

### ISSUE THREAD

**Issue by RTH216** [PhilippC/keepass2android#1906]
Since the upgrade to Android 12, I can't unlock the database using my correct password and key file.

I keep getting an error message stating "Invalid composite key" (German message "Ungültiger zusammengesetzter Schlüssel!")
On my PC I can open the database with key file and password.
I can open another database on Android 12 secured with just a password.
The key file is not broken  and saved on the smartphone itself. 

With Android 11 everything was working fine. 

I am using Android 12 (latest update provided by Vodafone/Samsung) on a Samsung Galaxy S20 ULTRA 
Keepass2android version 1.09a-r3

Any advice how to gain access to database again?

**Comment by PhilippC** [comment]
duplicate of https://github.com/PhilippC/keepass2android/issues/1824

**Comment by tops-br** [comment]
Hi everyone... I hope you're all well and that I can help you a little...

My scenario:
01 - A database created in Windows version 2.4x;
02 - This database is shared through a folder on my OneDrive, where the folder has write permission;
03 - Three Windows clients (versions 10 and 11, both Pro), running KeePass version 2.60 x64;
04 - 2 Linux Debian x64 clients running KeePass2 (2.57+dfsg-4);
05 - Three Android clients, one 16, one 15, and one 10, running Keepass2Android, version 1.14-r4 currently, but were on version 1.12.

All clients worked fine, logging in, sharing, and synchronizing the same database in real time, until, out of nowhere, problems started logging in, as everyone here is reporting.
After a lot of research, a lot indeed, without reaching any conclusion, I decided to test changing the database password using the "Change Master Key" option (File, Change Master Key). Initially, I changed it on a Windows workstation, but then I changed it again from Linux and Android as well, and now it's working correctly again.
